drivers/adc/adc_smartbond_gpadc.c

```diff
@@ -101,6 +110,44 @@ static int adc_smartbond_channel_setup(const struct device *dev,
 	return 0;
 }
 
+static inline void gpadc_smartbond_pm_policy_state_lock_get(const struct device *dev,
+					      struct adc_smartbond_data *data)
+{
+#if defined(CONFIG_PM_DEVICE_RUNTIME)
+	pm_device_runtime_get(dev);
+
+#endif
+
+#if defined(CONFIG_PM_DEVICE)
+	if (!atomic_test_and_set_bit(data->pm_policy_state_flag, 0)) {
+		/*
+		 * Prevent the SoC from entering the normal sleep state.
+		 */
+		pm_policy_state_lock_get(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
+	}
+
+#endif
```

Comment:
I thought that we don't need to do both.
```suggestion
#if defined(CONFIG_PM_DEVICE)
#if defined(CONFIG_PM_DEVICE_RUNTIME)
	pm_device_runtime_get(dev);
#else
	if (!atomic_test_and_set_bit(data->pm_policy_state_flag, 0)) {
		/*
		 * Prevent the SoC from entering the normal sleep state.
		 */
		pm_policy_state_lock_get(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
	}
#endif
#endif
```

Reply:
No, we need both.
So it should be:
```suggestion
#if defined(CONFIG_PM_DEVICE)
#if defined(CONFIG_PM_DEVICE_RUNTIME)
	pm_device_runtime_get(dev);
#endif
	if (!atomic_test_and_set_bit(data->pm_policy_state_flag, 0)) {
		/*
		 * Prevent the SoC from entering the normal sleep state.
		 */
		pm_policy_state_lock_get(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
	}
#endif
```

Reply:
We need both as in either case the system should not enter the sleep state; there is no PDC entry to notify the application core once the requested conversions are done.

Reply:
I applied @ydamigos suggestion for this code